apps/trading-e2e/src/support/pages/base-page.js

```diff
@@ -0,0 +1,56 @@
+export default class BasePage {
+  porfolioUrl = '/portfolio';
+  marketsUrl = '/markets';
+  connectVegaBtn = 'connect-vega-wallet';
+  walletConnectors = 'connectors-list';
+  walletForm = 'rest-connector-form';
+  walletFormError = 'form-error';
+
+  navigateToPortfolio() {
+    cy.get(`a[href='${this.porfolioUrl}']`).click();
+  }
+
+  navigateToMarkets() {
+    cy.get(`a[href='${this.marketsUrl}']`)
+      .should('be.visible')
+      .click({ force: true });
+    cy.url().should('include', '/markets');
+  }
+
+  navigateToConnectVegaWallet() {
+    cy.getByTestId(this.connectVegaBtn).click();
+    cy.contains('Connects using REST to a running Vega wallet service');
+    cy.getByTestId(this.walletConnectors).find('button').click();
+  }
+
+  fillInWalletForm(walletName, walletPassphrase) {
+    cy.getByTestId(this.walletForm)
+      .find('#wallet')
+      .click({ force: true })
+      .type(walletName);
+    cy.getByTestId(this.walletForm)
+      .find('#passphrase')
+      .click({ force: true })
+      .type(walletPassphrase);
+  }
+
+  clickConnectVegaWallet() {
+    cy.getByTestId(this.walletForm)
+      .find('button[type=submit]')
+      .click({ force: true });
+  }
+
+  validateWalletNotRunningError() {
+    cy.getByTestId(this.walletFormError).should(
+      'have.text',
+      'Authentication failed'
+    );
+  }
+
+  validateWalletErrorFieldsDisplayed() {
+    cy.getByTestId(this.walletForm)
+      .find('div > div')
```

Comment:
This selector might be a bit too prone to breaking if anyone edits the form. Lets add a data-testid or maybe `role='alert'` attribute to the InputError component